addons/sale/models/sale_order.py

```diff
@@ -706,6 +706,12 @@ def action_draft(self):
     def action_quotation_send(self):
         """ Opens a wizard to compose an email, with relevant mail template loaded by default """
         self.ensure_one()
+        for line in self.order_line.filtered(lambda order_line: not order_line.display_type and order_line.state in ['draft', 'sent']):
+            line._validate_analytic_distribution(**{
+                'product': line.product_id.id,
+                'business_domain': 'sale_order',
+                'company_id': line.company_id.id,
+            })
```

Comment:
> Why are we not only validating on Confirm, but also on Send?

From TSB, the reason is that when you send a sale order to a client, it can be validated by the client (when he accepts your SO), so it's at that moment, that you want the check.
For PO, you received it, and you will be the person to validate, so we don't need to check on the Send.